test_project/store/src/main/java/test/sergeyvalyushko/store/handlers/ExitApp.java

```diff
@@ -0,0 +1,19 @@
+package test.sergeyvalyushko.store.handlers;
+
+import org.xml.sax.SAXException;
+import test.sergeyvalyushko.store.Catalog;
+
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+
+public class ExitApp extends AbstractHandler{
+    public ExitApp(AbstractHandler nextHandler) {super(nextHandler);}
+
+    @Override
+    public void handleRequest(String input, Catalog catalog) throws ParserConfigurationException, SAXException, IOException {
+        if (input.equals("exit")){
+            System.exit(0);
+        }
+        else System.out.println("Incorrect input");
```

Comment:
also, not very cool implementation, what will be if i reordered commands initialization... so, you should interrupt execution if you did not find next handler/